### PR TITLE
Fix for memory leak (issue #33)

### PR DESCRIPTION
**Pull request by maximinus**
Fixes leak of PyObject in DukPy_eval_string (issue #33).

### CODE REVIEW

File: src/pyduktape.c [hunk 1]
```diff
@@ -44,6 +44,8 @@ static PyObject *DukPy_eval_string(PyObject *self, PyObject *args) {
     }
 
     ctx = get_context_from_capsule(pyctx);
+    Py_XDECREF(pyctx);
```
Comment (amol-) [inline src/pyduktape.c]: I think we should decrement the reference at the end of the function (on exit points).
The `pyctx` will call `duk_destroy_heap` through `context_destroy` when the references are gone, and thus will destroy the `ctx` which are are using below.

As far as we use `ctx` then we should not decrement `pyctx`. So I think `Py_XDECREF(pyctx)` should only go before a `return` within this method.

Comment (maximinus) [reply src/pyduktape.c]: Would you like me to post a new pull request tomorrow for this?

Comment (amol-) [reply src/pyduktape.c]: Yes please, it would be great if you can update the PR according to the suggestion 👍

**Comment by amol-** [comment]
I think what @webknjaz was suggesting is to test with something like:

```
gc.collect()
objects = gc.get_objects()
dukpy.evaljs(...)
gc.collect()
after_objects = gc.get_objects()
assert after_objects == objects
```

I'm fine with accepting this without a test unit btw, I'd love if you want to open another PR with the test

**Comment by maximinus** [comment]
I'll have to look into it more. From what I investigated I test like that will work even with the bug present. The issue is gc cannot see the memory leak (if it could see it, then it would be collected and there would be no bug). If I manage to find a way, I'll add a pull request.

**Comment by amol-** [comment]
It should see it, you should get the parcel listed in `gc.get_objects` and should never get away from there because it figures as there is a referece using it forever. At least in theory, didn't try it :D

**Comment by webknjaz** [comment]
I think that'd figure out only stuff that is exposed to Python. Internal leaks on the C-side would be left unnoticed.

**Comment by maximinus** [comment]
I did try to take the approach from above, but there was no difference in gc counts with leaking code and non-leaking code. It would probably need a re-compiled Python to unit test this one.

**Comment by webknjaz** [comment]
I think it's a good idea to to compile Python via pyenv in a separate job for this purpose.

**Pull request merged**